### ISSUE TITLE
Webfont usage within Sass (flickering)

### ISSUE THREAD

**Issue by zslabs** [gatsbyjs/gatsby#9826]
## Summary

This one is a bit sporadic so stay with me 😝 

I write all of my styles in `.scss`. After reading https://www.gatsbyjs.org/docs/adding-images-fonts-files/ I decided with the webfonts I needed to load I'd stick them in `/static/fonts/` and import them with Sass; no problem there at all. Developing locally, everything works fine and as expected. 

I noticed something odd though when building a production-version of the app. When hovering over links (that looks like it prefetches data) the webfonts would flicker on the page. Super strange stuff. I first thought they were corrupted files, but when converting them to base64 equivalents and not importing, the flickering stopped.

Fast forward to today, I decided to give this a fresh-go with newly generated webfonts from the foundries that do not have this issue outside of Gatsby, and the flickering continued. It's only when I imported the files within my base layout; and created some inline styles with react-helmet to add in the @font-face declarations was I able to import the font files and not get the flickering.

Could it be the prefetching that's causing the issue? I'm pretty certain it's not the files themselves, but it's only within Gatsby that this is an issue - as other React projects works fine (based off of create-react-app).

**Comment by zslabs** [comment]
I also tried just putting a static `.css` file with those font-face rules in the JS imports; and it's still doing the flickering. What only seems to work is a static `<head>` inclusion of those styles.

**Comment by jpalmieri** [comment]
We were also having this issue and seem to have been able to resolve it by adding a static css file and using `gatsby-browser.js` to import that file globally, e.g. `import './src/styles/global.css'`

**Comment by zslabs** [comment]
That’s super strange! I tried the same and could have sworn it worked initially, and then it went back to flickering. It’s partially a relief to know I’m not crazy over here 😜

**Comment by liorbrauer** [comment]
+1 On this issue, had the same thing happen, but only for links on pages that were under a subdirectory of `/pages`. (So basically all links to `/pages/careers/..` caused the reloading of fonts and flickering)

**Comment by sarneeh** [comment]
This is still a pain. It would be cool to be able to declare font-faces in `GlobalStyle`.

**Comment by patrikarvidsson** [comment]
I encountered the same issue when reading the fonts from `.sass`, but when moving `@font-face` to a `.css` and importing it in `gatsby-browser.js`  instead the problem is resolved.

In my case this only happened to `/about/`, for some weird reason.

**Comment by zslabs** [comment]
I "think" this could be fixed as of https://github.com/gatsbyjs/gatsby/pull/11800

**Comment by ezeikel** [comment]
I use `styled-components` so didn't want to have global css file, like the above suggested fix. Instead I included this in my `GlobalStyle.js` file:

```
// Hide fonts until webfonts have loaded to avoid FOUT
.wf-loading {
  visibility: hidden;
}
```

**Comment by wardpeet** [comment]
is this still an issue? It should have been fixed with #11800 our apologies.
